sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java

```diff
@@ -313,12 +320,24 @@ public synchronized void onStarted(@NonNull Context context, @NonNull Channel ch
         mAuthListener = new AbstractTokenContextListener() {
 
             @Override
+            // TODO check synchronized
+            // TODO post to background thread?
             public synchronized void onNewUser(String authToken) {
                 if (mLatestPushToken != null) {
                     enqueuePushInstallationLog(mLatestPushToken);
                 }
             }
         };
+        mUserListener = new UserIdContext.Listener() {
+
+            @Override
+            public void onNewUserId(String userId) {
+                // TODO use userId
```

Comment:
Probably pass it as param to enqueue to avoid querying context again (could have race condition).